src/super_gradients/training/losses/yolox_loss.py

```diff
@@ -0,0 +1,468 @@
+
+        en = (tl < br).type(tl.type()).prod(dim=1)
+        area_i = torch.prod(br - tl, 1) * en
+        area_u = area_p + area_g - area_i
+        iou = (area_i) / (area_u + 1e-16)
+
+        if self.loss_type == "iou":
+            loss = 1 - iou ** 2
+        elif self.loss_type == "giou":
+            c_tl = torch.min((pred[:, :2] - pred[:, 2:] / 2), (target[:, :2] - target[:, 2:] / 2))
+            c_br = torch.max((pred[:, :2] + pred[:, 2:] / 2), (target[:, :2] + target[:, 2:] / 2))
+            area_c = torch.prod(c_br - c_tl, 1)
+            giou = iou - (area_c - area_u) / area_c.clamp(1e-16)
+            loss = 1 - giou.clamp(min=-1.0, max=1.0)
+
+        if self.reduction == "mean":
+            loss = loss.mean()
+        elif self.reduction == "sum":
+            loss = loss.sum()
+
+        return loss
+
+
+class YoloXDetectionLoss(_Loss):
+    """
+    Calculate YOLO V5 loss:
+    L = L_objectivness + L_iou + L_classification + 1[no_aug_epoch]*L_l1
+    """
+
+    def __init__(self, strides, im_size, num_classes, use_l1=False, center_sampling_radius=2.5, iou_type='iou'):
+        super().__init__()
+        self.grids = [torch.zeros(1)] * len(strides)
+        self.strides = strides
+        self.num_classes = num_classes
+
+        self.center_sampling_radius = center_sampling_radius
+        self.use_l1 = use_l1
+        self.l1_loss = nn.L1Loss(reduction="none")
+        self.bcewithlog_loss = nn.BCEWithLogitsLoss(reduction="none")
+        self.iou_loss = IOUloss(reduction="none", loss_type=iou_type)
+
+    def forward(self, model_output, targets):
+        if isinstance(model_output, tuple) and len(model_output) == 2:
+            # in test/eval mode the Yolo v5 model output a tuple where the second item is the raw predictions
+            _, predictions = model_output
+        else:
+            predictions = model_output
+
+        return self.compute_loss(predictions, targets)
+
+    @staticmethod
+    def _make_grid(nx=20, ny=20):
+        yv, xv = torch.meshgrid([torch.arange(ny), torch.arange(nx)])
+        return torch.stack((xv, yv), 2).view((1, 1, ny, nx, 2)).float()
+
+    def compute_loss(self, predictions: List[torch.Tensor], targets: torch.Tensor) \
+            -> Tuple[torch.Tensor, torch.Tensor]:
+        """
+        L = L_objectivness + L_iou + L_classification + 1[no_aug_epoch]*L_l1
+        where:
+            * L_boxes, L_classification and L_l1 are calculated only between cells and targets that suit them;
+            * L_objectivness is calculated for all cells.
+
+        L_classification:
+            for cells that have suitable ground truths in their grid locations add BCEs
+            to force a prediction of IoU with a GT in a multi-label way
+            Coef: 1.
+        L_iou:
+            for cells that have suitable ground truths in their grid locations
+            add (1 - IoU^2), IoU between a predicted box and each GT box, force maximum IoU
+            Coef: 1.
```

Comment:
looks like coef is actually 5.0 (line215)

Reply:
fixed

Reply:
fixed